src/DIRAC/WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -345,12 +346,77 @@ def export_rescheduleJob(self, jobIDs):
     self.__sendJobsToOptimizationMind(validJobList)
     return result
 
-  def __deleteJob(self, jobID):
-    """ Delete one job
+  types_removeJob = []
 
-        :param int jobID: job ID
+  def export_removeJob(self, jobIDs):
+    """
+    Completely remove a list of jobs, also from TaskQueueDB,
+    and including its JobLogging info.
+    Only authorized users are allowed to remove jobs.
 
-        :return: S_OK()/S_ERROR()
+    :param list jobIDs: list of job IDs
+    :return: S_OK()/S_ERROR() -- confirmed job IDs
+    """
+
+    jobList = self.__getJobList(jobIDs)
+    if not jobList:
+      return S_ERROR('Invalid job specification: ' + str(jobIDs))
+
+    validJobList, invalidJobList, nonauthJobList, _ = self.jobPolicy.evaluateJobRights(jobList,
+                                                                                       RIGHT_DELETE)
+    count = 0
+    error_count = 0
+
+    self.log.verbose("Removing jobs", "(n=%d)" % len(validJobList))
+    result = self.jobDB.removeJobFromDB(validJobList)
+    if not result['OK']:
+      self.log.error("Failed to remove jobs from JobDB", "(n=%d)" % len(validJobList))
+    else:
+      self.log.info("Removed jobs from JobDB", "(n=%d)" % len(validJobList))
+
+    for jobID in validJobList:
+      resultTQ = self.taskQueueDB.deleteJob(jobID)
+      if not resultTQ['OK']:
+        self.log.warn("Failed to remove job from TaskQueueDB",
+                      "(%d): %s" % (jobID, resultTQ['Message']))
+        error_count += 1
+      else:
+        count += 1
+
+    result = self.jobLoggingDB.deleteJob(validJobList)
+    if not result['OK']:
+      self.log.error("Failed to remove jobs from JobLoggingDB", "(n=%d)" % len(validJobList))
+    else:
+      self.log.info("Removed jobs from JobLoggingDB", "(n=%d)" % len(validJobList))
+
+    if count > 0 or error_count > 0:
+      self.log.info("Removed jobs from DB",
+                    "(%d jobs with %d errors)" % (count, error_count))
+
+    if invalidJobList or nonauthJobList:
+      self.log.error(
+          "Jobs can not be removed",
+          ": %d invalid and %d in nonauthJobList" % (len(invalidJobList), len(nonauthJobList)))
+      errMsg = "Some jobs failed removal"
+      res = S_ERROR()
+      if invalidJobList:
+        self.log.debug("Invalid jobs: %s" % ','.join(str(ij) for ij in invalidJobList))
+        res['InvalidJobIDs'] = invalidJobList
+        res['Message'] = errMsg + ": invalid jobs"
```

Comment:
if invalidJobList is empty `errMsg` wouldn't be part of res['Message']

Reply:
Thanks, fixed.